libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.java

```diff
@@ -868,6 +893,113 @@ public void startSwitcherActivityIfRequired() {
         }
 	}
 
+    /**
+     * Updates developer support features to match an Android Activity lifecycle
+     * event, including updating the app's authenticated Android Activity.
+     *
+     * @param authenticatedActivity The new authenticated Android Activity
+     *                              instance, or null
+     * @param lifecycleActivity     The Android Activity instance provided by
+     *                              the Activity lifecycle event
+     */
+    private void updateDeveloperSupportForActivityLifecycle(
+            @Nullable Activity authenticatedActivity,
+            @NonNull Activity lifecycleActivity
+    ) {
+
+        // Show the developer support dialog, if applicable.
+        if (lifecycleActivity.isDestroyed() && lifecycleActivity instanceof ShowDeveloperSupportActivity) {
+            showDevSupportDialog(authenticatedActivityForDeveloperSupport);
+            return;
+        }
```

Comment:
It is an interesting part of the flow since the developer support dialog can theoretically be shown by any activity in the app - but as a dialog attached to any activity not as a new activity.  "Actions" on the notification need to start an activity to trigger the dismissal of the system views that display the notification or have a broadcast intent attached to the content intent of the notification.  The content intent recently got an inspector warning since Material design prefers an activity intent attached to an action instead of using the notification's content intent - especially a broadcast intent.  It's not a deprecation, though, so I believe it's acceptable to suppress if the design is required.

On one hand, I did like that using a dedicated activity doesn't require suppressing the content intent inspector warning.  On the other, I agree it's counterintuitive.

I filed a follow-up we can use to investigate using the content intent and see if it looks better.   Let's take a look and see which option we elect.  Thanks for the thoughts!

[W-13985181: [MSDK Android] Remove Unneeded Show Developer Support Activity
](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001YxPZ9YAN/view)

Reply:
...And here's the [follow-up code](https://github.com/JohnsonEricAtSalesforce/SalesforceMobileSDK-Android/pull/1#pullrequestreview-1587687813)